pkg/ast/pipesearch/wsSearchHandler.go

```diff
@@ -374,7 +374,7 @@ func createRecsWsResp(qid uint64, sizeLimit uint64, searchPercent float64, scrol
 		}
 	case structs.RRCCmd:
 		useAnySegKey := false
-		if aggs.OutputTransforms != nil && aggs.OutputTransforms.MaxRows != 0 {
+		if aggs.OutputTransforms != nil && (aggs.OutputTransforms.HeadRequest != nil && (aggs.OutputTransforms.HeadRequest.BoolExpr == nil && aggs.OutputTransforms.HeadRequest.MaxRows != 0)) {
```

Comment:
I think can be refactored to 
```
if aggs.OutputTransforms != nil && aggs.HasHeadBlock() && aggs.OutputTransforms.HeadRequest.MaxRows != 0 {
  // If block code.
}
```

Since you already have that method `hasHeadBlock` make it public and you can it use it here.

Reply:
this we cannot do since hasHeadBlock we can have a corner case like BoolExpr=true and maxrows is 3, in this case we would still need to continue processing the segments and not wait for search to be 100% finished (if we enter this if block it checks if search < 100 we break out).